pkg/indexer/atlasfederatedauthsecret.go

```diff
@@ -0,0 +1,44 @@
+package indexer
+
+import (
+	"go.uber.org/zap"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	akov2 "github.com/mongodb/mongodb-atlas-kubernetes/v2/pkg/api/v1"
+)
+
+const (
+	AtlasFederatedAuthBySecretsIndex = "atlasfederatedauth.spec.connectionSecret"
```

Comment:
Similar to the other PR, dataFed copied the connection secret implementation from deployment and also relied on matching DBUsers against the project. It would really benefit an index to the project. But now I believe it should be done in a separate PR, to also deduplicate the connection secret implementation.

Reply:
yep, agreed, this should be a separate PR.